rust, rustup: Use MSVC by default

comp500 opened the pull request:
Changes rust and rustup packages to use MSVC toolchain by default, as this is generally the better supported (and default) choice.

In the rustup package, the MSVC version is always used (the GCC build of rustup provides no benefits, and the MSVC version can be used to install GCC toolchains). The --default-host flag has been removed, so new installs will use the rustup default (MSVC) but existing installs will have a configuration file indicating GCC is the default; a note was added to the rustup packages stating how to change this.

The rust-gnu package was added, to allow using the GCC toolchain without rustup, and a note was added to all rust packages indicating that rustup can be used to more easily manage multiple toolchains (and is the primary install method; see https://forge.rust-lang.org/infra/other-installation-methods.html).

Closes #931.

comp500 commented:
/verify

rasa commented:
Whitespace issues:
https://github.com/ScoopInstaller/Main/actions/runs/3688451634/jobs/6245761114#step:4:29

rashil2000 commented:
The `rust-msvc` manifest can be removed (as it is identical to `rust`).

For rustup, we can have two files: `rustup` and `rustup-gnu`. Both files download the same file, but the `-gnu` manifest used the additional flag specifying the triplet while installation.

niheaven commented:
Add `rustup-gnu` and keep the two `-msvc` ones for back-compatibility.

rashil2000 commented:
We can make a breaking change now as the linked issue has been present for more than two years, so no need for back-compat IMO 🙂

niheaven commented:
I mean, someone is using rustup-msvc or rust-msvc (like me)...

rashil2000 commented:
Having all 3 variants will cause confusion honestly.

rustup-msvc and rust-msvc should be easily replaceable in your environment as they are independent.

niheaven commented:
Keep them until we have some solutions, IMO. Not everyone checks the update log or uses `scoop status` and `scoop info` to get changes.

comp500 commented:
I've added the `rustup-gnu` package defaulting to the GCC toolchain; because the default host is set in the install command it will be reset on every install/update of the package, so I added a note to make this clear to users.

niheaven commented:
/verify
